detection-rules/link_dynamics_form.yml

```
@@ -0,0 +1,30 @@
+name: "Link: Microsoft Dynamics 365 form phishing"
+description: |
+  Email body is suspicious, and links to a Microsoft Dynamics form. Known phishing tactic.
+references:
+  - "https://cofense.com/blog/compromised-microsoft-dynamic-365-customer-voice-account-used-for-phishing-attack/"
+type: "rule"
+severity: "high"
+source: |
+  type.inbound
+  and any(ml.nlu_classifier(strings.concat(
+          body.html.display_text, body.plain.raw)).intents, .name != "benign"
+  )
+  and (
+      any(body.links, .href_url.domain.domain == "ncv.microsoft.com")
+      or any(body.links,
+          beta.linkanalysis(.).effective_url.domain.domain == "customervoice.microsoft.com"
+          // confirm it is a form
+          and any(beta.linkanalysis(.).final_dom.links, .href_url.domain.domain == "cdn.forms.office.net")
+          and any(file.explode(beta.linkanalysis(.).screenshot),
+              any(ml.nlu_classifier(.scan.ocr.raw).intents,
+                  .name == "cred_theft" and
+                  .confidence == "high"
+              )
+          )
+      )
+  )
```

```suggestion
  )
  
  // unsolicited
  and (
          (
              sender.email.domain.root_domain in $free_email_providers
              and sender.email.email not in $recipient_emails
          )
          or (
              sender.email.domain.root_domain not in $free_email_providers
              and sender.email.domain.domain not in $recipient_domains
          )
  )
```

i think this should help reduce the number of messages we have to pass to NLU + LA, and hopefully not impact efficacy?